front/src/modules/ui/right-drawer/components/RightDrawerTopBar.tsx

```diff
@@ -15,19 +19,36 @@ const StyledRightDrawerTopBar = styled.div`
   font-size: ${({ theme }) => theme.font.size.md};
   gap: ${({ theme }) => theme.spacing(1)};
   height: 56px;
-  justify-content: flex-start;
+  justify-content: space-between;
+  margin-right: ${({ theme }) => theme.spacing(2)};
   padding-left: ${({ theme }) => theme.spacing(2)};
 
   padding-right: ${({ theme }) => theme.spacing(2)};
 `;
 
+const TopBarWrapper = styled.div`
+  display: flex;
+`;
+
 export function RightDrawerTopBar() {
   const isMobile = useIsMobile();
+  const activityId = useRecoilValue(viewableActivityIdState);
+
+  const { data } = useGetActivityQuery({
```

Comment:
I feel we should not query the activity in the RightDrawerTopBar. I actually feel we don't need it here as we only need the activityId which we already have.
If we would need the activity, I would, create a <LoadActivityData> Component that would do the fetch call and store the activity in recoil.
use this recoil state in RightDrawerTopBar and other RightDrawerComponents that need to get access to the activity